### PR TITLE
[VEN-2611] Set Redstone as the Main Oracle for BNB and BTC

### PR DESCRIPTION
**Pull request by web3rover**
## Description

Resolves VEN-2611

### CODE REVIEW

**Review by chechu — APPROVED** [review]

File: simulations/vip-306/simulations.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  testVip("Update BTC and BNB Price Config", vip303(60 * 60 * 24 * 7), {
+    callbackAfterExecution: async (txResponse: TransactionResponse) => {
+      await expectEvents(txResponse, [REDSTONE_ORACLE_ABI], ["TokenConfigAdded"], [4]);
+      await expectEvents(txResponse, [BOUND_VALIDATOR_ABI], ["ValidateConfigAdded"], [2]);
+      await expectEvents(txResponse, [RESILIENT_ORACLE_ABI], ["TokenConfigAdded"], [2]);
+    },
+  });
+
+  describe("Post-VIP state", () => {
+    it("validate price", async () => {
```
Comment (kkirka) [inline simulations/vip-306/simulations.ts:41]: I'd feel safer if we had failure simulations here as well, e.g. RedStone and Chainlink reporting different prices. BNB and BTC are quite huge markets, and we'd rather not have any "invalid price" errors there, especially on a volatile market when we need liquidations to happen. Wdyt?

Comment (chechu) [reply simulations/vip-306/simulations.ts:41]: We have this: https://github.com/VenusProtocol/oracle/pull/194

**Review by kkirka — APPROVED** [review]

File: vips/vip-306/bscmainnet.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+export const BNB_FEED_REDSTONE = "0x8dd2D85C7c28F43F965AE4d9545189C7D022ED0e";
+export const BTC_FEED_REDSTONE = "0xa51738d1937FFc553d5070f43300B385AA2D9F55";
+
+export const BNB_FEED_CHAINLINK = "0x0567F2323251f0Aab15c8dFb1967E4e8A7D42aeE";
+export const BTC_FEED_CHAINLINK = "0x264990fbd0A4796A3E3d8E37C4d5F87a3aCa5Ebf";
+
+const vip306 = (maxStalePeriod: number) => {
+  const meta = {
+    version: "v2",
+    title: "VIP-304 Set RedStone as the Main Oracle for BTC and BNB",
```
Comment (Debugger022) [inline vips/vip-306/bscmainnet.ts:28]:
```suggestion
    title: "VIP-306 Set RedStone as the Main Oracle for BTC and BNB",
```

File: simulations/vip-306/simulations.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { TransactionResponse } from "@ethersproject/providers";
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { expectEvents } from "../../src/utils";
+import { forking, testVip } from "../../src/vip-framework";
+import vip303, { BNB, BTC, RESILIENT_ORACLE } from "../../vips/vip-306/bscmainnet";
```
Comment (Debugger022) [inline simulations/vip-306/simulations.ts:9]:
```suggestion
import vip306, { BNB, BTC, RESILIENT_ORACLE } from "../../vips/vip-306/bscmainnet";
```

File: simulations/vip-306/simulations.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    it("validate price", async () => {
+      const bnbPrice = await oracle.getPrice(BNB);
+      expect(bnbPrice).to.equal(parseUnits("620.30", 18));
+
+      const btcPrice = await oracle.getPrice(BTC);
+      expect(btcPrice).to.equal(parseUnits("70979.20075", 18));
+    });
+  });
+
+  testVip("Update BTC and BNB Price Config", vip303(60 * 60 * 24 * 7), {
```
Comment (Debugger022) [inline simulations/vip-306/simulations.ts:32]:
```suggestion
  testVip("Update BTC and BNB Price Config", vip306(60 * 60 * 24 * 7), {
```

**Review by Debugger022 — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/310?chainId=1

**Pull request merged**